src/os/open_sockets.rs

```diff
@@ -0,0 +1,58 @@
+use ::std::collections::HashMap;
+
+use ::sysinfo::ProcessExt;
+use netstat2::*;
+
+use crate::network::{LocalSocket, Protocol};
+use crate::OpenSockets;
+use sysinfo::{Pid, System, SystemExt};
+
+pub(crate) fn get_open_sockets() -> OpenSockets {
+    let mut open_sockets = HashMap::new();
+
+    let mut sysinfo = System::new_all();
+    sysinfo.refresh_processes();
+
+    let af_flags = AddressFamilyFlags::IPV4 | AddressFamilyFlags::IPV6;
+    let proto_flags = ProtocolFlags::TCP | ProtocolFlags::UDP;
+    let sockets_info = get_sockets_info(af_flags, proto_flags);
+
+    if let Ok(sockets_info) = sockets_info {
+        for si in sockets_info {
+            let mut procname = "";
```

Comment:
Can we make this a `String::new()`? I think this will make things a little clearer in the code forks below, and I think we'll end up with the same number of allocations.

Reply:
Sure thing, that was more of a C/C# legacy than optimalization :)